src/Streamly/Internal/Data/Unfold.hs

```diff
@@ -475,6 +486,35 @@ repeatM = Unfold step return
     {-# INLINE_LATE step #-}
     step x = return $ Yield x x
 
+{-# INLINE iterateM #-}
+iterateM :: Monad m => (a -> m a) -> Unfold m a a
```

Comment:
Would you like to add documentation for `iterateM`, `generateM`and `fromIndicesM`?

Reply:
@adithyaov